### PR TITLE
Where to start looking page

### PR DESCRIPTION
**Pull request by matt-lethargic**
Implemented the happy path as I understand it for Where To Start Looking page.
This page will need some slight changes when the pages after it are in

### CODE REVIEW

File: cypress/integration/WhereStartLookingPage.feature [hunk 1]
```diff
@@ -12,3 +12,13 @@ Feature: Where should we start looking Page
...
+
+  Scenario: Selecting One Country along with an active country and clicking check eligibility navigates Relative In Hiding page
+    When I select "one country in particular" radio button
+    When I click the "check eligibility" button
+    Then I am on the "relatives in hiding" page
+
+  Scenario: Selecting Two or Three Countries clicking check eligibility navigates Relative In Hiding page
+    When I select "two or three countries" radio button
+    When I click the "check eligibility" button
+    Then I am on the "relatives in hiding" page
```
Comment (WillGibson) [inline cypress/integration/WhereStartLookingPage.feature]: The prototype does this...

image

I'm wondering why we need the one country versus 2 or 3 if the result is the same. Why don't we just let them select up to three?

File: src/routes/handlers/director.handler.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import * as express from 'express';
```
Comment (WillGibson) [inline src/routes/handlers/director.handler.ts]: I would argue that you don't need the `.handler` in the file name as it's implicit (almost explicit) in the file path `src/routes/handlers/director.ts`.

Should we unit test this?

Comment (WillGibson) [reply src/routes/handlers/director.handler.ts]: Or maybe comment that it is not unit tested because it's covered by the Cypress tests?

Comment (matt-lethargic) [reply src/routes/handlers/director.handler.ts]: Let's not argue! I don't mind either way. 
Was thinking about the unit tests, would be good, what framework?

File: cypress/integration/WhereStartLookingPage.feature [hunk 3]
```diff
@@ -12,3 +12,14 @@ Feature: Where should we start looking Page
   Scenario: Has a back link to navigate to the how did you lose contact page
     When I click the "back" link
     Then I am on the "how did you lose contact with your family" page
+
+  Scenario: Selecting one or more active Countries and clicking check eligibility navigates Relative In Hiding page
```
Comment (WillGibson) [inline cypress/integration/WhereStartLookingPage.feature]:
```suggestion
  Scenario: Selecting one or more active Countries and clicking check eligibility navigates to Relative In Hiding page
```

File: cypress/integration/WhereStartLookingPage.feature [hunk 4]
```diff
@@ -12,3 +12,14 @@ Feature: Where should we start looking Page
...
+    When I select "one to three countries" radio button
+    When I select "AL" in "country-1" select box
+    When I click the "check eligibility" button
+    Then I am on the "relatives in hiding" page
+
+  Scenario: Selecting Countries radio button and no countries and clicking check eligibility does not navigate
```
Comment (WillGibson) [inline cypress/integration/WhereStartLookingPage.feature]:
```suggestion
  Scenario: Clicking check eligibility does nothing when selecting Countries radio button, but no countries.
```

File: cypress/integration/WhereStartLookingPage.feature [hunk 5]
```diff
@@ -12,3 +12,14 @@ Feature: Where should we start looking Page
   Scenario: Has a back link to navigate to the how did you lose contact page
     When I click the "back" link
     Then I am on the "how did you lose contact with your family" page
+
+  Scenario: Selecting one or more active Countries and clicking check eligibility navigates Relative In Hiding page
+    When I select "one to three countries" radio button
+    When I select "AL" in "country-1" select box
```
Comment (WillGibson) [inline cypress/integration/WhereStartLookingPage.feature]:
```suggestion
    When I select "AL" in the "country-1" select menu
```

Can we use the human friendly country name in here too?

File: cypress/integration/common/Content.js [hunk 6]
```diff
@@ -36,3 +36,8 @@ When(/^I see a "([^"]*)" button linking to the "([^"]*)" page$/, function (butto
 When(/^I select "([^"]*)" radio button$/, function (radioText) {
   cy.get('[type="radio"]').check(textToValue[radioText]);
 });
+
+When(/^I select "([^"]*)" in "([^"]*)" select box$/, function (value, selectId) {
+  const selectBox = cy.get(`#${selectId}`);
+  selectBox.select(value);
```
Comment (WillGibson) [inline cypress/integration/common/Content.js]:
```suggestion
  const selectBox = cy.get(`#${selectId}`).select(value);
```

File: src/routes/handlers/director.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import * as express from 'express';
+
```
Comment (WillGibson) [inline src/routes/handlers/director.ts:2]:
```suggestion

// Not unit tested as the behaviour is covered by the Cypress UI tests
```

File: src/routes/handlers/where-start-looking.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+const fs = require('fs');
+const path = require('path');
+const viewData = {
+  countries: allCountries
+};
+
+export function whereGet(req: express.Request, res: express.Response) {
+  res.render('where-start-looking', viewData);
+}
+
```
Comment (WillGibson) [inline src/routes/handlers/where-start-looking.ts:13]:
```suggestion

// Not unit tested as the behaviour is covered by the Cypress UI tests
```

File: src/views/where-start-looking.html [hunk 9]
```diff
@@ -31,18 +30,76 @@
...
+                    </div>
+                  </div>
+                </fieldset>
+              </div>
 
-            {{ redcrossButton({
+              {{ redcrossButton({
                         text: "check eligibility",
                         attributes: { id: "check-eligibility-button" }
                     }) }}
```
Comment (WillGibson) [inline src/views/where-start-looking.html]: Anal retention, sorry :-)
```suggestion
              {{ redcrossButton({
                text: "check eligibility",
                attributes: { id: "check-eligibility-button" }
              }) }}
```

**Review by stevehenshaw — APPROVED** [review]

**Review by WillGibson — APPROVED** [review]

**Pull request merged**